modules/homa-lite/src/lib.rs

```diff
@@ -914,14 +915,14 @@ pub mod module {
 		}
 
 		/// Helper function that update the storage with the new
-		fn update_total_staking_currency_storage(new_total: Balance) -> DispatchResult {
-			ensure!(!new_total.is_zero(), Error::<T>::InvalidTotalStakingCurrency);
-
-			// Update storage and emit event.
-			TotalStakingCurrency::<T>::mutate(|current| {
-				*current = new_total;
-				Self::deposit_event(Event::<T>::TotalStakingCurrencySet(*current));
+		fn update_total_staking_currency_storage(f: impl FnOnce(Balance) -> Balance) -> DispatchResult {
```

Comment:
```suggestion
		fn update_total_staking_currency_storage(f: impl FnOnce(Balance) -> Result<Balance, DispatchError>) -> DispatchResult {
```

and then use try_mutate so the closure can return error